src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilTest.java

```diff
@@ -75,6 +79,21 @@ public void testIsCheckstyleModule() {
                 .isTrue();
     }
 
+    /**
+    * This test case is designed to verify the behavior of getCheckstyleModules method.
+    * It is provided with a package name that does not contain any checkstyle modules.
+    * It ensures that ModuleReflectionUtil.getCheckstyleModules is returning an empty set.
+    */
```

Comment:
@mahfouz72 it might be good to make sure that @romani is good with this spending time updating this test. If this is the route we take, please mention your research here:
 - exactly which part of the code we are hitting with this test (the filter operation)
 - why we couldn't make any normal test fail
 - why we didn't want to remove the code (we don't want to load everything in the class path)

Reply:
okay I am waiting for @romani approval before updating the comment

Reply:
@romani ping

Reply:
please consider https://github.com/checkstyle/checkstyle/pull/14499/files#r1499361210

Reply:
I am confirming that making Input based test is possible but will be big hack.
patch to get to call of this method:
```
 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java b/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java
index 1124219..9c6815b 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java
@@ -60,7 +60,7 @@ public final class ModuleReflectionUtil {
         return packages.stream()
                 .flatMap(pkg -> classPath.getTopLevelClasses(pkg).stream())
                 .map(ClassPath.ClassInfo::load)
-                .filter(ModuleReflectionUtil::isCheckstyleModule)
+                //.filter(ModuleReflectionUtil::isCheckstyleModule)
                 .collect(Collectors.toUnmodifiableSet());
     }
 
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java b/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
index 0b9de5c..f45ed14 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
@@ -179,7 +179,8 @@ public final class InlineConfigParser {
             // Inlined config is not supported for non java files.
             "com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck",
             "com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck",
-            "com.puppycrawl.tools.checkstyle.checks.TranslationCheck"
+            "com.puppycrawl.tools.checkstyle.checks.TranslationCheck",
+        "LineSeparatorOption"
     );
 
     /** Stop instances being created. **/
@@ -318,6 +319,8 @@ public final class InlineConfigParser {
                 "com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck");
         moduleMappings.put("SummaryJavadoc",
                 "com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck");
+        moduleMappings.put("LineSeparatorOption",
+                "LineSeparatorOption");
         if (moduleMappings.containsKey(moduleName)) {
             fullyQualifiedClassName = moduleMappings.get(moduleName);
         }
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/InputFinalParameters.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/InputFinalParameters.java
index a69b300..636a960 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/InputFinalParameters.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/InputFinalParameters.java
@@ -1,5 +1,5 @@
 /*
-FinalParameters
+LineSeparatorOption
 ignorePrimitiveTypes = (default)false
 tokens = (default)METHOD_DEF, CTOR_DEF
```